libs/libmysqlxx/include/mysqlxx/PoolFactory.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <memory>
+#include "PoolWithFailover.h"
+
+#define MYSQLXX_POOL_WITH_FAILOVER_DEFAULT_START_CONNECTIONS 1
+#define MYSQLXX_POOL_WITH_FAILOVER_DEFAULT_MAX_CONNECTIONS 16
+#define MYSQLXX_POOL_WITH_FAILOVER_DEFAULT_MAX_TRIES 3
+
+namespace mysqlxx
+{
+/*
+ * PoolFactory.h
+ * This class is a helper singleton to mutualize connections to MySQL.
+ */
+class PoolFactory final : public ext::singleton<PoolFactory>

```

Comment:
PoolFactory must be made thread-safe even if currently its usage is always serialized.